### PR TITLE
Added networkMode 

### PR DESCRIPTION
**Pull request by masgari**
More control over --net options (none,bridge,host,container)
```
task createContainer(type: DockerCreateContainer) {
    imageId = 'imageId'
    containerName = 'name'
    networkMode = 'host'
}  
```

### CODE REVIEW

**Comment by bmuschko** [comment]
Could you please create a new functional test for `DockerCreateContainer`? It should create a new container with this is option and afterward remove it.

**Comment by masgari** [comment]
I couldn't run all functional tests with Docker 1.8.3 due to NPE in class `InspectContainerResponse` of docker-java client. Only tested networkMode with `gradle fT --tests *networkMode*` 

Can be fixed in `InspectContainerResponseHandler` by catching exception for `$container.volumes`.

**Comment by bmuschko** [comment]
Thanks for providing the PR. Unfortunately, your functional test is failing.

    :connectServer FAILED
    :connectServer (Thread[Daemon worker,5,main]) completed. Took 0.032 secs.

    BUILD FAILED

    Total time: 2.742 secs

    -----
    Error:
      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                     Dload  Upload   Total   Spent    Left  Speed

      0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to localhost port 27277: Connection refused

    FAILURE: Build failed with an exception.

    * What went wrong:
    Execution failed for task ':connectServer'.
    > Process 'command 'curl'' finished with non-zero exit value 7

Could you please look into it?

**Comment by bmuschko** [comment]
@masgari We'll need to fix the test before we can merge your pull request. Would you mind looking into it?

File: src/functTest/groovy/com/bmuschko/gradle/docker/DockerWorkflowFunctionalTest.groovy [hunk 1]
```diff
@@ -226,6 +226,67 @@ class DockerWorkflowFunctionalTest extends AbstractFunctionalTest {
...
+            }
+
+            task startContainer(type: DockerStartContainer) {
+              dependsOn createContainer
+              targetContainerId { createContainer.getContainerId() }
+            }
+
+            task connectServer(type: Exec) {
+              dependsOn startContainer
+              commandLine 'curl', 'http://localhost:27277'
```
Comment (cdancy) [inline src/functTest/groovy/com/bmuschko/gradle/docker/DockerWorkflowFunctionalTest.groovy]: Is there no way we can get the localhost and port in a more dynamic fashion?

**Comment by masgari** [comment]
Sorry for my late reply. I've tested it on Ubuntu and inside a clean vagrant vm and it is working.

Could you provide more information about your docker version, OS and how you run the tests please?

**Comment by cdancy** [comment]
@masgari you can infer some information from the Snap CI output here: https://snap-ci.com/bmuschko/gradle-docker-plugin/pull/114

**Comment by bmuschko** [comment]
@masgari Some [info on executing the tests](https://github.com/bmuschko/gradle-docker-plugin#executing-the-plugins-test-suite-with-custom-configuration). Simply run `./gradlew build` and it will execute unit, integration and functional tests.

**Comment by bmuschko** [comment]
Thanks for the pull request. It has been merged. I removed the test as it wasn't working properly. The assumption here is that Docker Java hopefully tested this functionality.

**Pull request merged**